okhttp/src/main/java/okhttp3/internal/ws/RealWebSocket.java

```diff
@@ -493,13 +541,20 @@ boolean writeOneFrame() throws IOException {
         writer.writePong(pong);
 
       } else if (messageOrClose instanceof Message) {
-        ByteString data = ((Message) messageOrClose).data;
+        ByteString uncompressedData = ((Message) messageOrClose).data;
+
+        ByteString data = uncompressedData;
+        if (messageDeflater != null) {
+          data = CompressionCorrector.applyPostDeflate(
```

Comment:
Combine CompressionCorrector and MessageDeflater. They’re useless without each other, right?

Reply:
Done, same for MessageInflater. CompressionCorrector is now gone.